R/derive_param_diam_to_sev.R

```diff
@@ -0,0 +1,226 @@
+#'
+#'
+#' @param mod limit for grade `"MODERATE"`
+#'
+#' *Default: c(5,10)*
+#' *Permitted value*: A Numeric vector
+#'
+#' @param sev limit for grade `"SEVERE"`
+#'
+#' *Default: 10*
+#' *Permitted value*: A Numeric vector
+#'
+#' @details
+#' 1. Pass the Input data set in `dataset` with required variables and `AVAL`
+#' should have the diameter values for the events(`FAOBJ`) specified in
+#' `filter_faobj`.
+#' 2. If the SDTM data set has severity record for redness and swelling, User
+#' can skip this function to keep the SDTM level severity records for further
+#'  derivation.
+#' 3. If User want to derive the severity records from the diameter records even
+#'  that data set has severity records for Redness and swelling. This function
+#'  will remove the existing severity records and it will create the new
+#'  severity records as`AVALC` which is derived from diameter. and `AVAL` will
+#'  be derived as numeric severity grade to get the maximum severity records
+#'  which will be derived by `derive_param_maxsev.R`.
+#' 4. Pass the values in severity grade arguments(`none`, `mild`, `mod`, `sev`)
+#'  as per the study needs.
+#' 5.User can pass the events which has diameter values to derive their severity
+#'  records by passing the appropriate `filter_diam`, `filter_faobj`and severity
+#'  grade limits.
+#'
+#' @return
+#' The Input data with the new severity records for Redness and swelling which
+#' is specified in `filter_faobj` and AVAL, AVALC will be derived and fatestcd,
+#'  fatest will be changed as per the values
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(admiral)
+#' library(admiraldev)
+#' library(tibble)
+#' Input <- tribble(
+#'   ~USUBJID, ~FAOBJ, ~AVAL, ~AVALC, ~ATPTREF,
+#'   "XYZ1001", "REDNESS", 7.5, 7.5, "VACCINATION 1",
+#'   "XYZ1001", "REDNESS", 3.5, 3.5, "VACCINATION 1",
+#'   "XYZ1001", "REDNESS", 2, 2, "VACCINATION 1",
+#'   "XYZ1001", "REDNESS", 1.8, 1.8, "VACCINATION 1",
+#'   "XYZ1001", "REDNESS", 1.4, 1.4, "VACCINATION 1",
+#'   "XYZ1002", "REDNESS", 11.1, 11.1, "VACCINATION 2",
+#'   "XYZ1002", "REDNESS", 7.4, 7.4, "VACCINATION 2",
+#'   "XYZ1002", "REDNESS", 6, 6, "VACCINATION 2",
+#'   "XYZ1002", "REDNESS", 2.1, 2.1, "VACCINATION 2",
+#'   "XYZ1002", "REDNESS", 1.1, 1.1, "VACCINATION 2",
+#'   "XYZ1001", "SWELLING", 5.5, 5.5, "VACCINATION 1",
+#'   "XYZ1001", "SWELLING", 2.5, 2.5, "VACCINATION 1",
+#'   "XYZ1001", "SWELLING", 2, 2, "VACCINATION 1",
+#'   "XYZ1001", "SWELLING", 1.8, 1.8, "VACCINATION 1",
+#'   "XYZ1001", "SWELLING", 1.4, 1.4, "VACCINATION 1",
+#'   "XYZ1002", "SWELLING", 10.1, 10.1, "VACCINATION 2",
+#'   "XYZ1002", "SWELLING", 7.1, 7.1, "VACCINATION 2",
+#'   "XYZ1002", "SWELLING", 5, 5, "VACCINATION 2",
+#'   "XYZ1002", "SWELLING", 1.8, 1.8, "VACCINATION 2",
+#'   "XYZ1002", "SWELLING", 1.4, 1.4, "VACCINATION 2"
+#' ) %>% mutate(
+#'   AVALC = as.character(AVALC),
+#'   FATEST = "Diameter",
+#'   FATESTCD = "DIAMETER"
+#' )
+#'
+#' output <- derive_param_diam_to_sev(
```

Comment:
```suggestion
#' derive_param_diam_to_sev(
```

Reply:
By not assigning to an object the example will print out the results/output.  This is nice for our website as you don't have to run the code to see the output.